bin/fin

```diff
@@ -5083,28 +5084,98 @@ addon () {
 
 	case "$1" in
 		install|in)
-				addon_install "$addon_name" ${is_global}
-			;;
+			addon_install "$addon_name" ${is_global}
+		;;
 		remove|rm)
 			addon_remove "$addon_name" ${is_global}
-			;;
+		;;
+		update|up)
+			addon_update "$addon_name" ${is_global}
+		;;
 		version|v)
-			# special case for addon version command
+		# special case for addon version command
 			addon_script=$(get_addon_script "$addon_name")
 			[ ! -f "$addon_script" ] && echo "-1" && exit
 			addon_version=$(cat "$addon_script" | grep ^VERSION | cut -d = -f 2 | tr -d \" | tr -d \')
 			addon_version="${addon_version:-0}"
 			echo "$addon_version"
-			;;
+		;;
 		help)
 			show_help_addon
-			;;
+		;;
 		*)
 			echo-yellow "Unknown sub-command. See ${NC}fin addon help${yellow} for details."
-			;;
+		;;
 	esac
 }
 
+# Check to see if addon exists within remote repo
+# $1 - uuid to use for foldername in /tmp
+# $2 - addon_name the folder/file to find
+# $3 - repo_url the repository url to check
+# $4 - repo_branch the branch to check against
+_addon_temp_install () {
+	local uuid=${1}
+	local addon_name=${2}
+	local repo_url=${3}
+	local repo_branch=${4}
+	local tmp_random="/tmp/${uuid}"
+	mkdir $tmp_random && cd $tmp_random
+
+	# Initialize empty repo; suppress output
+	git init > /dev/null
+	git remote add origin $repo_url > /dev/null
+	git fetch -q origin $repo_branch > /dev/null
+	git cat-file -s origin/${repo_branch}:${addon_name} 2> /dev/null
+	git checkout origin/${repo_branch} ${addon_name} 2> /dev/null
+}
+
+# Search all of the provided repositories for the requested plugin.
+_addon_search_repo () {
+	eval 'repos=(${!'"DOCKSAL_ADDONS_REPO"'@})'
+
+	local addon=$1
+	local repo_url=$2
+	local repo_branch=$3
+	local repo_uuid=$4
+
+	echo-green "Searching for ${addon} in predefined repositories..."
+	for key in "${repos[@]}"
+	do
+		local config=${!key}
```

Comment:
no need for this, it should be `for config in "${repos[@]}"`

Reply:
@achekulaev 

`repos` is an array of variable names
When looping through those variables, `key` refers to the variable name that is being used.
`config` is referring to the value for the variable that is stored.

Example:

```bash
DOCKSAL_ADDONS_REPO="https://github.com/docksal/addons"
DOCKSAL_ADDONS_REPO_X="https://github.com/kanopi/docksal-toolkit.git|sample"
eval 'repos=(${!'"DOCKSAL_ADDONS_REPO"'@})'
# should produce something like the following array
# repos=(DOCKSAL_ADDONS_REPO DOCKSAL_ADDONS_REPO_X)
for key in "${repos[@]}"
do
	local config=${!key} #when looped will now grab the value for that variable
done
```

Reply:
Right. Then let's refactor to 
`for repo in "${repos[@]}"`
key is just too generic.